specs/bep-v1.rst

```diff
@@ -60,10 +60,86 @@ as noted per message type - any message type may be sent at any time and
 the sender need not await a response to one message before sending
 another.
 
-The underlying transport protocol MUST be TCP.
+The underlying transport protocol MUST guarantee reliable packet delivery.
 
-Messages
---------
+Pre-authentication messages
+---------------------------
+
+AFTER establishing a connection, but BEFORE performing any authentication,
+*devices* MUST exchange Hello messages.
+
+Hello messages are used to carry additional information about the peer, which
+might be of interest to the user even if the peer is not permitted to
+communicate due to failing authentication.
+
+Hello messages MUST be prefixed with a magic number **0x9F79BC40**
+represented in network byte order (BE), followed by 4 bytes representing the
+size of the message in network byte order (BE), followed by the content of
+the Hello message itself. The content of Hello message MUST be less or equal
+to 1024 bytes.
+
+::
+
+    Prefix Structure:
+
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                             Magic                             |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                            Length                             |
```

Comment:
I know the code is already running, but it's the way of things that they appear clearer when in the specification than in code :)